test/rcheevos/test_trigger.c

```diff
@@ -1940,6 +1940,121 @@ static void test_large_memref_not_shared() {
+  assert_hit_count(trigger, 1, 0, 0U);
+  assert_hit_count(trigger, 2, 2, 1U);
+
+  /* core group condition is true - hit count should be incremented */
+  /* alt group condition is false since it's a different accumulator */
+  /* alt2 group condition is true - hit count should be incremented */
+  assert_evaluate_trigger(trigger, &memory, 0);
+  assert_hit_count(trigger, 0, 1, 2U);
+  assert_hit_count(trigger, 1, 0, 0U);
+  assert_hit_count(trigger, 2, 2, 2U);
+
+  /* core group condition is true - hit count should be incremented to reach target */
+  /* alt group condition is false since it's a different accumulator */
+  /* alt2 group condition is true - hit count should be incremented */
+  assert_evaluate_trigger(trigger, &memory, 0);
+  assert_hit_count(trigger, 0, 1, 3U);
+  assert_hit_count(trigger, 1, 0, 0U);
+  assert_hit_count(trigger, 2, 2, 3U);
+
+  /* core group condition is true - target previously met */
+  /* alt group condition is false since it's a different accumulator */
+  /* alt2 group condition is true - hit count should be incremented */
+  assert_evaluate_trigger(trigger, &memory, 0);
+  assert_hit_count(trigger, 0, 1, 3U);
+  assert_hit_count(trigger, 1, 0, 0U);
+  assert_hit_count(trigger, 2, 2, 4U);
+
+  /* core group condition is true - target previously met */
+  /* alt group condition is false since it's a different accumulator */
+  /* alt2 group condition is true - hit count incremented to reach target */
+  /* core + alt2 now satisfied, trigger is true*/
+  assert_evaluate_trigger(trigger, &memory, 1);
+  assert_hit_count(trigger, 0, 1, 3U);
+  assert_hit_count(trigger, 1, 0, 0U);
+  assert_hit_count(trigger, 2, 2, 5U);
+}
+
+static void test_remember_recall_use_same_value_multiple() {
+  uint8_t ram[] = { 0x00, 0x12, 0x34, 0xAB, 0x56 };
+  memory_t memory;
+  rc_trigger_t* trigger;
+  char buffer[512];
+
+  memory.ram = ram;
+  memory.size = sizeof(ram);
+
+  ram[0] = 1;
+  assert_parse_trigger(&trigger, buffer, "E:5_A:0xH00_C:{accumulator}=6_B:0xH00_C:{accumulator}=4_M:0=1.4.");
+
+  /* because of accumulator can be re-used, both add hits are true and increment hits */
+  assert_evaluate_trigger(trigger, &memory, 0);
+  assert_hit_count(trigger, 0, 2, 1U);
+  assert_hit_count(trigger, 0, 4, 1U);
+  ASSERT_NUM_EQUALS(trigger->measured_value, 2U);
+  ASSERT_NUM_EQUALS(trigger->measured_target, 4U);
+
+  /* because of accumulator can be re-used, both add hits are true and increment hits to reach target*/
+  assert_evaluate_trigger(trigger, &memory, 1);
+  assert_hit_count(trigger, 0, 2, 2U);
+  assert_hit_count(trigger, 0, 4, 2U);
+  ASSERT_NUM_EQUALS(trigger->measured_value, 4U);
+  ASSERT_NUM_EQUALS(trigger->measured_target, 4U);
+
+  /* condition is true - previously met */
+  assert_evaluate_trigger(trigger, &memory, 1);
+  assert_hit_count(trigger, 0, 2, 3U);
+  assert_hit_count(trigger, 0, 4, 3U);
+  ASSERT_NUM_EQUALS(trigger->measured_value, 6U);
+  ASSERT_NUM_EQUALS(trigger->measured_target, 4U);
+}
+
```

Comment:
I would also like to see the following tests:
* Something chaining the accumulator (remember accumulator * value)
* Something using multiple accumulators in the same group (remember, do something, remember something else, do something else)
* Something using multiple accumulators in the same group where one is used by a Pause condition and one is used by a normal condition. (remember, pause, remember something else, do something else)
  - also pause second (remember something else, do something else, remember, pause)
* Something using the accumulator without setting it first.

Reply:
I see some of these might be covered in the value tests below.

Reply:
Yeah, some cases were functionally tested in the value tests which I added first. For those that are should I  include coverage in trigger as well? May be some merit. I'll review to add uncovered cases next time I work on this.

Reply:
I think I have coverage for all of these in either test_trigger.c or test_value.c now. If I missed one, let me know.